### PR TITLE
Remove Menus code

### PR DESCRIPTION
**Pull request by kwight**
This is cleanup from #12967 , where the Menus link was replaced by links to the Customizer.

To test:
* Restart Calypso
* Make sure that it still works as before.
* Do some `grep`ping for `menus` to make sure we've found all occurrences.

Last open question:
* ~Do we also want to remove menus related stuff from `wpcom-undocumented`?~ Removed.

### CODE REVIEW

File: client/my-sites/menus/controller.js [hunk 1]
```diff
@@ -1,77 +0,0 @@
...
-					secondaryActionURL: site.options.admin_url + 'nav-menus.php',
-					secondaryActionTarget: '_blank'
-				} )
-			),
-			document.getElementById( 'primary' ),
-			context.store
-		);
-	}
-
-	if ( site && site.jetpack && ! site.hasJetpackMenus ) {
```
Comment (tyxla) [inline client/my-sites/menus/controller.js]: This also removes the last usage of the `hasJetpackMenus` computed attribute. This means you can safely remove:

```
this.hasJetpackMenus = versionCompare( this.options.jetpack_version, '3.5-alpha' ) >= 0;
```
from `JetpackSite.prototype.updateComputedAttributes`.

**Comment by ockham** [comment]
Blocked by https://github.com/Automattic/wp-calypso/issues/12980.

**Comment by ockham** [comment]
Do we also want to remove menus related stuff from `wpcom-undocumented`?

File: assets/stylesheets/shared/functions/_z-index.scss [hunk 2]
```diff
@@ -84,8 +83,6 @@ $z-layers: (
 		'.people-list-item .card__link-indicator': 2,
 		'.updated-confirmation': 2,
 		'.auth__form .form-fieldset input:focus': 2,
-		'.menus__menu-item': 2,
-		'.menus__menu-item-open:before': 2,
```
Comment (gwwar) [inline assets/stylesheets/shared/functions/_z-index.scss]: 🎉

**Pull request merged**